### ISSUE TITLE
ClickHouse dialect does not support GLOBAL keyword

### ISSUE THREAD

**Issue by damirbk** [sqlfluff/sqlfluff#5256]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

ClickHouse has a keyword `GLOBAL` that can be used to [change the behaviour of distributed subqueries](https://clickhouse.com/docs/en/sql-reference/operators/in). GLOBAL IN and GLOBAL JOIN can be used. It seems like SQLFluff doesn't handle this, and reports a parsing error.

### Expected Behaviour

GLOBAL should be supported by the SQLFluff ClickHouse dialect.

### Observed Behaviour

A parsing error is reported.

### How to reproduce

test.sql:
```sql
SELECT uniq(UserID)
FROM distributed_table
WHERE
    CounterID = 101500
    AND UserID GLOBAL IN (SELECT UserID FROM distributed_table WHERE CounterID = 34)
```
The SQL is taken from the [documentation on the IN operator](https://clickhouse.com/docs/en/sql-reference/operators/in), so it should work.

`sqlfluff lint test.sql --dialect=clickhouse` returns the following:
```
== [test.sql] FAIL                                                                                                                                                      
L:   5 | P:  15 |  PRS | Line 5, Position 15: Found unparsable section: ' GLOBAL
                       | IN (SELECT UserID FROM distribut...'
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
```
SQLFluff is also complaining about CP02 (capitalisation.identifiers) but that's not relevant to this problem.

### Dialect

ClickHouse

### Version

SQLFluff version 2.3.2
Python version 3.10.10

### Configuration

```
[sqlfluff]
dialect = clickhouse
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
Could you please give a `GLOBAL JOIN` query to test on?

**Comment by damirbk** [comment]
Sure, sorry about that!

The [ClickHouse join docs](https://clickhouse.com/docs/en/sql-reference/statements/select/join) have this specification for joins:

```sql
SELECT <expr_list>
FROM <left_table>
[GLOBAL] [INNER|LEFT|RIGHT|FULL|CROSS] [OUTER|SEMI|ANTI|ANY|ASOF] JOIN <right_table>
(ON <expr_list>)|(USING <column_list>) ...
```

So it's just a matter of adding GLOBAL before the join specification keywords (inner/left/etc). Example from the docs where I've added GLOBAL:

```
SELECT name, text
FROM table_1
GLOBAL LEFT OUTER JOIN table_2
    ON table_1.Id = table_2.Id
    AND startsWith(table_2.text, 'Text')
```
